plugins/modules/ise_radius_integration_workflow_manager.py

```diff
@@ -734,366 +739,372 @@ def get_want_authentication_policy_server(self, auth_policy_server):
 
-            shared_secret = str(shared_secret)
-            if len(shared_secret) < 4 or len(shared_secret) > 100:
-                self.msg = "The 'shared_secret' should contain between 4 and 100 characters."
-                self.status = "failed"
-                return self
+            auth_server_exists = self.have.get("authenticationPolicyServer")[auth_server_index].get("exists")
 
-            invalid_chars = " ?<"
-            for char in invalid_chars:
-                if char in shared_secret:
-                    self.msg = "The 'shared_secret' should not contain spaces or the characters '?', '<'."
+            if not auth_server_exists:
+                shared_secret = item.get("shared_secret")
+                if not shared_secret:
+                    self.msg = "Missing parameter 'shared_secret' is required."
                     self.status = "failed"
                     return self
 
-            auth_server.update({"sharedSecret": shared_secret})
-
-        protocol = auth_policy_server.get("protocol")
-        if protocol not in ["RADIUS", "TACACS", "RADIUS_TACACS", None]:
-            self.msg = "protocol should either be ['RADIUS', 'TACACS', 'RADIUS_TACACS']." + \
-                       "It should not be {0}".format(protocol)
-            self.status = "failed"
-            return self
+                shared_secret = str(shared_secret)
+                if len(shared_secret) < 4 or len(shared_secret) > 100:
+                    self.msg = "The 'shared_secret' should contain between 4 and 100 characters."
+                    self.status = "failed"
+                    return self
 
-        if protocol is not None:
-            auth_server.update({"protocol": protocol})
-        else:
-            if not auth_server_exists:
-                auth_server.update({"protocol": "RADIUS"})
-            else:
-                auth_server.update({"protocol": auth_server_details.get("protocol")})
+                invalid_chars = " ?<"
+                for char in invalid_chars:
+                    if char in shared_secret:
+                        self.msg = "The 'shared_secret' should not contain spaces or the characters '?', '<'."
+                        self.status = "failed"
+                        return self
 
-        auth_server.update({"port": 49})
+                auth_server.update({"sharedSecret": shared_secret})
 
-        if not auth_server_exists:
-            encryption_scheme = auth_policy_server.get("encryption_scheme")
-            if encryption_scheme not in ["KEYWRAP", "RADSEC", None]:
-                self.msg = "The encryption_scheme should be in ['KEYWRAP', 'RADSEC']. " + \
-                           "It should not be {0}.".format(encryption_scheme)
+            protocol = item.get("protocol")
+            if protocol not in ["RADIUS", "TACACS", "RADIUS_TACACS", None]:
+                self.msg = "protocol should either be ['RADIUS', 'TACACS', 'RADIUS_TACACS']." + \
```

Comment:
self.msg = (
        "The 'protocol' should be one of ['RADIUS', 'TACACS', 'RADIUS_TACACS'], "
        "not '{0}'.".format(protocol)
    )